clippy_lints/src/utils/author.rs

```diff
@@ -190,43 +191,300 @@ impl PrintVisitor {
+                pat_, arg_pat, body_pat, self.current
+            );
+
+            self.current = pat_;
+            self.visit_pat(pat);
+
+            self.current = arg_pat;
+            self.visit_expr(arg);
+
+            self.current = body_pat;
+            self.visit_expr(body);
+
+            return true;
+        }
+
+        false
+    }
+
+    fn print_match_expr(&mut self, expr: &Expr<'_>, arms: &[Arm<'_>], des: MatchSource) {
+        let expr_pat = self.next("expr");
+        let arms_pat = self.next("arms");
+
+        println!(
+            "    if let ExprKind::Match(ref {}, ref {}, MatchSource::{:?}) = {}.kind;",
+            expr_pat, arms_pat, des, self.current
+        );
+
+        self.current = expr_pat;
+        self.visit_expr(expr);
+
+        println!("    if {}.len() == {};", arms_pat, arms.len());
+
+        for (i, arm) in arms.iter().enumerate() {
+            self.current = format!("{}[{}].body", arms_pat, i);
+            self.visit_expr(arm.body);
+
+            if let Some(ref guard) = arm.guard {
+                let guard_pat = self.next("guard");
+
+                println!("    if let Some(ref {}) = {}[{}].guard;", guard_pat, arms_pat, i);
+
+                match guard {
+                    hir::Guard::If(if_expr) => {
+                        let if_expr_pat = self.next("expr");
+
+                        println!("    if let Guard::If(ref {}) = {};", if_expr_pat, guard_pat);
+
+                        self.current = if_expr_pat;
+                        self.visit_expr(if_expr);
+                    },
+                    hir::Guard::IfLet(if_let_pat, if_let_expr) => {
+                        let if_let_pat_pat = self.next("pat");
+                        let if_let_expr_pat = self.next("expr");
+
+                        println!(
+                            "    if let Guard::IfLet(ref {}, ref {}) = {};",
+                            if_let_pat_pat, if_let_expr_pat, guard_pat
+                        );
+
+                        self.current = if_let_expr_pat;
+                        self.visit_expr(if_let_expr);
+
+                        self.current = if_let_pat_pat;
+                        self.visit_pat(if_let_pat);
+                    },
+                }
+            }
+            self.current = format!("{}[{}].pat", arms_pat, i);
+            self.visit_pat(arm.pat);
+        }
+    }
 }
 
-struct PrintVisitor {
+struct PrintVisitor<'a, 'tcx> {
     /// Fields are the current index that needs to be appended to pattern
     /// binding names
     ids: FxHashMap<&'static str, usize>,
     /// the name that needs to be destructured
     current: String,
+    cx: &'a LateContext<'tcx>,
 }
 
-impl<'tcx> Visitor<'tcx> for PrintVisitor {
+impl<'a, 'tcx> Visitor<'tcx> for PrintVisitor<'a, '_> {
     type Map = Map<'tcx>;
 
     #[allow(clippy::too_many_lines)]
     fn visit_expr(&mut self, expr: &Expr<'_>) {
-        print!("    if let ExprKind::");
```

Comment:
Is there a particular reason to change this?

Reply:
Nope, forgot to add that back after adding `check_higher`. Thanks :)